qutebrowser/qutebrowser
# Feat/mac sandbox pre release pyinstaller

Switch to PyInstaller develop branch to get v6.0 changes which brings:

* WebEngine/Chromium sandboxing for macOS builds
* Removes the need for our downstream patching of the app on macOS
* Unifies resource lookups to be via importlib_resources (this is actually possible as of 2021 but we hadn't picked up on it)

This is a pretty standard adaptation to a new major version of a dependency except in this case the major version isn't released yet, so I'm pointing at a specific commit on their develop branch (the latest commit as of today).

In my opinion we probably want to do a little more testing of the resulting binaries before committing to this. Although I don't think the testing surface area is huge? Like it runs, resource files can be loaded, it can be signed? There are binaries [here](https://github.com/qutebrowser/qutebrowser/actions/runs/5678007969), based of [this](https://github.com/qutebrowser/qutebrowser/tree/feat/test_pre_release_pyinstaller) test branch with a custom CI job.

Potential testing checklist:

* [ ] installs over an old version with no warnings
* [ ] it launches
* [ ] can prompt to capture audio or location
    * test with jitsi meet
* [ ] :help is rendered like normal
* [ ] userscripts work (if packaged?)
* [ ] pdf-js works
    * https://www.ipcc.ch/site/assets/uploads/2021/06/Fact_sheet_AR6.pdf
    * or qute://pdfjs/build/pdf.js on builds with 5.15.2
* [ ] application icon on shortcut and taskbar is correct
* [ ] extensions are loaded
    * the reload command is registered


TODO:
* open issue to track reverting the temp stuff for the pre-release pyinstaller once a release comes out (requirements files and PYINSTALLER_COMPILE_BOOTLOADER) :heavy_check_mark:  #7806
* probably move the PYINSTALLER_COMPILE_BOOTLOADER=true out of tox into the build script so that it gets set for local builds too :heavy_check_mark: 
* do manual testing I guess. I went through a few earlier, but not all of them. :heavy_check_mark: 
* remove the warning page and the place that triggers it qutebrowser/html/warning-sandboxing.html :heavy_check_mark: 
* apologies to @rokm for pulling in the develop branch

Should I, GitHub, should I? :thinking: 
![image](https://github.com/qutebrowser/qutebrowser/assets/7419144/2cace295-151f-4c8d-8585-11a954f5dbfb)

Testing (it seems userscripts are not packaged):

Windows (10)

* [x] installs over an old version with no warnings
* [x] it launches
* [x] can prompt to capture audio or location
      * tested with jitsi meet
* [x] :help is rendered like normal
* [x] pdf-js works
      * https://www.ipcc.ch/site/assets/uploads/2021/06/Fact_sheet_AR6.pdf
* [x] application icon on shortcut and taskbar is correct
* [x] extensions are loaded
      * the reload command is registered

macOS (11.7.8)

* [x] installs over an old version with no warnings
* [x] it launches
* [x] can prompt to capture audio or location
      * tested with jitsi meet
* [x] :help is rendered like normal
* [x] pdf-js works
      * https://www.ipcc.ch/site/assets/uploads/2021/06/Fact_sheet_AR6.pdf
* [x] application icon on shortcut and taskbar is correct
* [x] extensions are loaded
      * the reload command is registered

Closes: #7278
